lib/engine/game/g_1828.rb

```diff
@@ -264,12 +282,25 @@ def coalfields.blocks?(corporation)
         end
       end
 
+      def can_run_route?(entity)
+        return false if entity.id == 'C&P' && !@round.last_tile_lay
+
+        super
+      end
+
+      def city_tokened_by?(city, entity)
+        return @graph.connected_nodes(entity).keys.include?(city) if entity.id == 'C&P'
```

Comment:
why would you do keys.include?

can't you just do connected_nodes(entity)[city]

Reply:
Good catch. Let's pretend that never happened lol.